app/Http/Controllers/SuggestPlansController.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Plan;
+use App\Models\Genre;
+use App\Models\Place;
+use App\Models\Keyword;
+use App\Models\Interest;
+use App\Models\Question;
+use App\Models\PlanDetail;
+use Illuminate\Http\Request;
+
+class SuggestPlansController extends Controller
+{
+    const DEFAULT_SPEND_TIME = 90;
+    const MOVE_TIME = 30;
+    const START_TIME = '09:00';
+
+    public function index(Request $request)
+    {
+        if (is_null($request->input('question_id'))) {
+            session(['question_no' => 1]);
+            session()->forget('answers');
+        } else {
+            session(['question_no' => session('question_no') + 1]);
+
+            // Dont include on select result
+            $question_id = $request->input('question_id');
+
+            if (!empty($request->input('answer'))) {
+                session(['answers.question_'.$question_id.'.question_id' => $question_id]);
+                foreach ($request->input('answer') as $answer) {
+                    session()->push('answers.question_'.$question_id.'.answer_id', $answer);
+                }
+            }
+        }
+
+        $interest = new Interest();
+        $interest_data = $interest->selectRandomInterest();
+
+        // Whether to use interest
+        if (is_null($interest_data)) {
+            $is_use_interest = false;
+        } else {
+            $r_use_interest_num = mt_rand(1, 3);
+            if ($r_use_interest_num == 1 || $r_use_interest_num == 2) {
```

Comment:
we should not use magic number.
I know we only have limited time,  if you can, pls fix it.
https://suzumura-tumiage.com/php/69/